lib/msal-browser/src/broker/nativeBroker/PlatformAuthProvider.ts

```diff
@@ -0,0 +1,151 @@
+    LogLevel,
+    Logger,
+    AuthenticationScheme,
+} from "@azure/msal-common/browser";
+import { name, version } from "../../packageMetadata.js";
+import {
+    BrowserConfiguration,
+    DEFAULT_NATIVE_BROKER_HANDSHAKE_TIMEOUT_MS,
+} from "../../config/Configuration.js";
+import { BrowserPerformanceClient } from "../../telemetry/BrowserPerformanceClient.js";
+import { NativeMessageHandler } from "./NativeMessageHandler.js";
+import { NativeConstants } from "../../utils/BrowserConstants.js";
+import { PlatformDOMHandler } from "./PlatformDOMHandler.js";
+
+export const PLATFORM_EXTENSION_PROVIDER = "NativeMessageHandler";
+export const PLATFORM_DOM_PROVIDER = "PlatformDOMHandler";
+
+export async function isPlatformBrokerAvailable(
+    loggerOptions?: LoggerOptions,
+    perfClient?: IPerformanceClient
+): Promise<boolean> {
+    const defaultLoggerOptions: LoggerOptions = {
+        loggerCallback: (): void => {
+            // Empty logger callback
+        },
+        piiLoggingEnabled: false,
+        logLevel: LogLevel.Trace,
+    };
+
+    const logger = new Logger(
+        loggerOptions || defaultLoggerOptions,
+        name,
+        version
+    );
+
+    logger.trace("isPlatformBrokerAvailable called");
+
+    const defaultPerformanceClientConfig = {
+        auth: {
+            clientId: "",
+        },
+    };
+
+    const performanceClient =
+        perfClient ||
+        new BrowserPerformanceClient(defaultPerformanceClientConfig);
+
+    if (!window) {
+        logger.trace("Non DOM environment detected, returning false");
+        return false;
+    }
+
+    // Check if DOM platform API is supported
+
+    // @ts-ignore
+    if (window.navigator?.platformAuthentication) {
+        const supportedContracts =
+            // @ts-ignore
+            await window.navigator.platformAuthentication.getSupportedContracts(
+                NativeConstants.MICROSOFT_ENTRA_BROKERID
+            );
+        if (supportedContracts.includes("get-token-and-sign-out")) {
+            logger.trace("Platform auth available in DOM");
+            return true;
+        }
+    }
+
+    // Check and initialize native extension provider if available
+    try {
+        const nativeExtensionProvider =
+            await NativeMessageHandler.createProvider(
+                logger,
+                DEFAULT_NATIVE_BROKER_HANDSHAKE_TIMEOUT_MS,
+                performanceClient
+            );
+        if (nativeExtensionProvider) {
+            logger.trace(
+                "Platform auth available via extension, returning true"
+            );
+            return true;
+        } else {
+            logger.trace("Platform auth not available, returning false");
+            return false;
+        }
+    } catch (e) {
+        logger.trace(e as string);
+        return false;
+    }
+}
+
+/**
+ * PlatformAuthProvider is a utility class that provides methods to check if the platform broker is available
+ */
+export class PlatformAuthProvider {
```

Comment:
Please don't make this a class. It prevents the function from being minified/treeshaked and it's static anyway

Reply:
+1